### PR TITLE
adds logic to add images for articles with recipes and without images

### PR DESCRIPTION
**Pull request by tsop14**
When ETL is run, after new articles are indexed...

- gets list of articles with recipes saved [A]
- gets list of articles with images saved [B]
- works out articles with missing images by diffing [A] and [B]

### CODE REVIEW

**Comment by shtukas** [comment]
LGTM, thanks ! :+1:

File: etl/src/main/scala/etl/ETL.scala [hunk 1]
```diff
@@ -68,6 +68,16 @@ object ETL extends App {
       println(s"Finished! End result: $endResult")
       println("Articles with no recipes:")
       endResult.articlesWithNoRecipes.foreach(id => println(s"- https://www.theguardian.com/$id"))
+
+      //add images for articles with recipes and no images
+      def articlesWithoutImages: Set[String] = db.getArticlesWithRecipes.toSet diff db.getArticlesWithSavedImages.toSet
+
+      articlesWithoutImages.foreach(a => {
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]:
```  
    articlesWithoutImages.foreach { articleId =>
         capiClient.getResponse(ItemQuery(articleId)).foreach { itemResponse =>
           db.insertImages(ImageExtraction.getImages(itemResponse.content).toList)
         }
       }
```

File: etl/src/main/scala/etl/ETL.scala [hunk 2]
```diff
@@ -132,7 +142,7 @@ object ETL extends App {
   def getRecipes(content: Content)(insertImages: List[ImageDB] => Unit): Seq[Recipe] = {
 
     val rawRecipes: Seq[RawRecipe] = RecipeExtraction.findRecipes(content.webTitle, content.fields.flatMap(_.body).getOrElse(""))
-    if (rawRecipes.nonEmpty) { insertImages(ImageExtraction.getImages(content, content.id).toList) }
+    if (rawRecipes.nonEmpty) { insertImages(ImageExtraction.getImages(Some(content)).toList) }
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]: Why does this need to be a Some? We already have a Content object.

File: etl/src/main/scala/etl/ImageExtraction.scala [hunk 3]
```diff
@@ -5,8 +5,11 @@ import com.gu.recipeasy.models.{ ImageDB }
 
 object ImageExtraction {
 
-  def getImages(content: Content, articleId: String): Seq[ImageDB] = {
-    findImageElements(content).flatMap(i => extractImageData(i, articleId))
+  def getImages(maybeContent: Option[Content]): Seq[ImageDB] = {
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ImageExtraction.scala]: Do you need an option as an argument here?

**Comment by shtukas** [comment]
@tsop14 
I have discussed with @LATaylor-guardian about his suggestions.

What happened is that in `db.insertImages(ImageExtraction.getImages(itemResponse.content).toList)` turns out that `itemResponse.content` is an option. So I was lazy and changed the signature of `getImages` to take an option, but then had to change how `getImages` is called in `insertImages(ImageExtraction.getImages(Some(content)).toList)`

File: etl/src/main/scala/etl/ETL.scala [hunk 4]
```diff
@@ -68,6 +68,16 @@ object ETL extends App {
       println(s"Finished! End result: $endResult")
       println("Articles with no recipes:")
       endResult.articlesWithNoRecipes.foreach(id => println(s"- https://www.theguardian.com/$id"))
+
+      //add images for articles with recipes and no images
+      def articlesWithoutImages: Set[String] = db.getArticlesWithRecipes.toSet diff db.getArticlesWithSavedImages.toSet
+
+      articlesWithoutImages.foreach( articleId => {
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]: When the foreach is over multiple lines it's 
```
.foreach { myVal => 
    ...
}
```

Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]: Normal brackets for oneliner and you don't need thd curly brace after the `=>`

**Comment by LATaylor-guardian** [comment]
👍

**Pull request merged**